kits/arm/arm.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+import hebi
+from time import sleep, time, perf_counter
+import numpy as np
+
+import os
+import sys
+
+# ------------------------------------------------------------------------------
+# Add the root folder of the repository to the search path for modules
+root_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))
+sys.path = [root_path] + sys.path
+# ------------------------------------------------------------------------------
+
+from util.math_utils import get_grav_comp_efforts
+from util.math_utils import gravity_from_quaternion
+
+class ArmParams(object):
+    
+    def __init__(self, family, moduleNames, hrdf):
+        self.family = family
+        self.hrdf = hrdf
+        self.moduleNames = moduleNames
+
+
+    
+class Arm():
+    
+    def __init__(self, params):
+        # Create arm from lookup
+        self.lookup = hebi.Lookup()
+        sleep(2)
```

Comment:
I really don't want to do this in the constructor... We should probably expose the Arm API as something like the following:

```py

def create_arm(params, lookup_sleep_amount=2, other_input_parameters...):
  lookup = hebi.Lookup()
  sleep(lookup_sleep_amount)
  group = lookup.get_group_from_names([params.family], params.module_names)
  if group is None:
    raise RuntimeError("Could not create group") # Probably want to raise exception here on failure IMO?
  return Arm(group) # Can add other constructor parameters as needed here too
```

Reply:
I ran into something similar in MATLAB and I'm a big fan of just passing in the group directly. The lookup has so many settings and overloads that it seems easier to just use the existing API rather than replicating it via factory methods.